presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -592,6 +600,40 @@ public TableStatisticsMetadata getStatisticsCollectionMetadata(Session session,
         return metadata.getStatisticsCollectionMetadata(session.toConnectorSession(connectorId), tableMetadata);
     }
 
+    @Override
+    public AnalyzeTableMetadata getAnalyzeTableMetadata(Session session, TableHandle tableHandle, Map<String, Object> properties)
+    {
+        ConnectorId connectorId = tableHandle.getConnectorId();
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, connectorId);
+        ConnectorMetadata metadata = catalogMetadata.getMetadata();
+
+        ConnectorAnalyzeTableMetadata analyzeTableMetadata = metadata.getAnalyzeTableMetadata(session.toConnectorSession(connectorId), tableHandle.getConnectorHandle(), properties);
+        return new AnalyzeTableMetadata(
+                analyzeTableMetadata.getStatisticsMetadata(),
+                new TableMetadata(connectorId, analyzeTableMetadata.getTableMetadata()),
+                new TableHandle(connectorId, analyzeTableMetadata.getTableHandle()));
+    }
+
+    @Override
+    public AnalyzeTableHandle beginAnalyze(Session session, TableHandle tableHandle)
+    {
+        ConnectorId connectorId = tableHandle.getConnectorId();
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, connectorId);
+        ConnectorMetadata metadata = catalogMetadata.getMetadata();
+
+        ConnectorTransactionHandle transactionHandle = catalogMetadata.getTransactionHandleFor(connectorId);
+        ConnectorAnalyzeTableHandle connectorTableHandle = metadata.beginAnalyze(session.toConnectorSession(connectorId), tableHandle.getConnectorHandle());
```

Comment:
you should be passing analyze properties to `metadata.beginAnalyze`

Reply:
Technically you can store whatever information you need in the TableHandle. Let's add properties when ones are needed. Currently the properties will be unused.

Reply:
OTOH you shouldn't be forced to squash everything into `TableHandle`. It's not a case class... I'm fine with not having properties here... It might be needed by someone adding ANALYZE in custom connector, but that's not me today.

Reply:
I'm reluctant of adding unused code, even if it is SPI. If we ever need this, we can change the SPI.

Reply:
I agree with @arhimondr here. We can change it when it's actually needed.